### PR TITLE
Push initial version

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
Heavily inspired in `erlef/setup-elixir`.

Nothing is guaranteed to work, as of now, which is why I'm opening as a draft.

Test are passing locally, though.

By myself:
- [x] generic: have all FIXME dealt with (this includes making sure `rebar3` is optionally(?) bundled)
- [x] `rebar3`: look in `erlef/setup-elixir` how different versions are being obtained (they aren't: we may download from GitHub releases, though)
- [x] `rebar3`: add a test project (something simple from a template should do, for now)
- [x] `licensed`: check what's up the gem installation and then update stuff and make CI green

With collaboration:
- [x] `setup-erlang.test.js`: potentially described and add more tests (needs to wait until the `rebar3` bit is done)
- [x] ~~`package.json`: decide on what the first version is (1.0.0?)~~ this decision is outside the scope of this pull request, but the value should be adjusted
- [x] downloading external packages: no checksum or signature validation is done. Should we care about this?

### CODE REVIEW

File: src/install-erlang [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+OS=${1}
+ERLVSN=${2}
+FILE=${ERLVSN}.tar.gz
+DIR_FOR_BIN=.setup-erlang/erlang
+
+wget -q https://repo.hex.pm/builds/otp/"${OS}"/"${FILE}"
+mkdir -p ${DIR_FOR_BIN}
+tar zxvf "${FILE}" -C ${DIR_FOR_BIN}
+rm -f "${FILE}"
+echo "$(pwd)/${DIR_FOR_BIN}/${ERLVSN}/bin" >> "$GITHUB_PATH"
```
Comment (paulo-ferraz-oliveira) [inline src/install-erlang:16]: This is still no good, because:
1. there is no `erl` here
2. the bin directory is inside `erts-...`

Comment (paulo-ferraz-oliveira) [reply src/install-erlang]: @ericmj gave me a hand here, and now it's better 😄

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  unit_test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - run: npm ci
+      - run: npm test
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/test.yml:20]: This bit's broken on `setup-elixir` (might be worth opening an issue there?). The fact is it doesn't return an error, which is why CI doesn't fail.

File: __tests__/setup-erlang.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const {getErlangVersion} = require('../src/setup-erlang')
```
[the remark this answers is not in the document]
Comment (paulo-ferraz-oliveira) [reply __tests__/setup-erlang.test.js:1]: Moving to an action on the description at the top.

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+name: Setup Erlang/OTP
+description: Set up a specific version of Erlang/OTP and add the command-line tools to the PATH
+author: GitHub
```
[the remark this answers is not in the document]
Comment (paulo-ferraz-oliveira) [reply action.yml]: I'll open an issue.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "setup-erlang",
+  "version": "0.1.0",
```
[the remark this answers is not in the document]
Comment (paulo-ferraz-oliveira) [reply package.json:3]: Moving to the top description.

File: .github/workflows/test.yml (hunk 2 above)
Comment (paulo-ferraz-oliveira) [reply .github/workflows/test.yml:25]: Will open an issue there.

**Comment by paulo-ferraz-oliveira** [comment]
Moving this away from draft as I'm nearing closure of initial version (I'm only worried about CI fixes now). @starbelly, would appreciate comments on open tasks.

File: README.md [hunk 6]
```diff
@@ -1,23 +1,26 @@
-# setup-erlang
+<!-- markdownlint-disable MD013 -->
+# setup-erlang [![GitHub Actions Test][test-img]][test] [![GitHub Actions CI][ci-img]][ci]
 
-[![](https://github.com/erlef/setup-erlang/workflows/Test/badge.svg)](https://github.com/erlef/setup-erlang/actions)
-[![](https://github.com/erlef/setup-erlang/workflows/Licensed/badge.svg)](https://github.com/erlef/setup-erlang/actions)
+[test]: https://github.com/erlef/setup-erlang
```
Comment (starbelly) [inline README.md:4]: TIL I can write links like this 👍

**Review by starbelly — APPROVED** [review]
I think this is a solid first version 💯 ❤️

**Comment by starbelly** [comment]
@paulo-ferraz-oliveira 

> setup-erlang.test.js: potentially described and add more tests (needs to wait until the rebar3 bit is done)

I think the level of tests, especially for an initial version is adequate. We ensure : 

  -  erlang is available
   - rebar3 is available
   - we check versions
   - We make sure we can run common rebar3 operations in a trivial test project.

I think that's solid for now. I can think of other things to test, but they should be discussed later, as they are probably out of scope and make for slower brittle tests. 

> downloading external packages: no checksum or signature validation is done. Should we care about this?

I think we should consider it, but I don't think it should be done as part of this PR. We also need to take into consideration if bob is be replaced with a github action, then we will likely be downloading from github, and if that's so then I'm not sure a sum check helps much, but it also doesn't hurt anything. None the less, I would do this in a subsequent PR myself.

File: README.md (hunk 6 above)
Comment (paulo-ferraz-oliveira) [reply README.md:4]: Thumbs up!

**Comment by paulo-ferraz-oliveira** [comment]
Task created: https://github.com/erlef/setup-erlang/issues/8.

**Comment by paulo-ferraz-oliveira** [comment]
We're missing nothing. I just want to refactor a very small function there and then I'll squash it all up.

**Comment by paulo-ferraz-oliveira** [comment]
It's been squashed.

**Comment by paulo-ferraz-oliveira** [comment]
I'm closing this in favour of https://github.com/erlef/setup-elixir/pull/9. @starbelly, shall we go ahead and move the issues there, already?

**Comment by starbelly** [comment]
@paulo-ferraz-oliveira yeah, let's do it!

**Comment by paulo-ferraz-oliveira** [comment]
> @paulo-ferraz-oliveira yeah, let's do it!

Done. I think the project can be archived/deleted, whatever you prefer.

**Comment by starbelly** [comment]
Let' delete it since there wasn't much here, will be less confusing for people too i think.

**Pull request closed without merging**